### PR TITLE
Removed unwanted spaces when converting bson to json string.

### PR DESCRIPTION
**Pull request by fun2C0d3**
I find no need for adding spaces in the created json string. Undesired when using this method to send a response to client.

### CODE REVIEW

File: src/bson/bson.c [hunk 1]
```diff
@@ -2681,7 +2681,7 @@ _bson_as_json_visit_bool (const bson_iter_t *iter,
 {
    bson_json_state_t *state = data;
 
-   bson_string_append (state->str, v_bool ? "true" : "false");
+   bson_string_append (state->str, v_bool ? "true":"false");
```
Comment (bjori) [inline src/bson/bson.c]: thats not a printed space

**Comment by bjori** [comment]
Although I do agree that these spaces don't add any value, I don't think we'll be removing them either.

File: src/bson/bson.c (hunk 1 above)
Comment (fun2C0d3) [inline src/bson/bson.c]: Correct. Seems I removed one to many spaces..
Would be great if the spaces were optional though. I assume most people using this method use it (like me) to generate a json string to send back to a client. Adding all this spaces for no reason makes the response larger without adding any functionality.

**Comment by fun2C0d3** [comment]
Would be great if the spaces were optional though. I assume most people using this method use it (like me) to generate a json string to send back to a client. Adding all this spaces for no reason makes the response larger without adding any functionality.

**Comment by bjori** [comment]
In practice, if the size of the document is important, wouldn't you compress the reply anyway?

That seems like it would safe you a whole lot more then removing few spaces?

**Comment by fun2C0d3** [comment]
Sure. I just don't get the reason for adding spaces when not needed, but it makes no big difference :)
Thanks for a good library anyways 👍

**Pull request closed without merging**